regions/regions.go

```diff
@@ -44,14 +45,34 @@ func (rg Regions) RoundTrip(req *http.Request) (*http.Response, error) {
 	}
 	shardsRing, ok := rg.multiCluters[reqHost]
 	if ok {
+		req.Header.Add(storage.HOST, reqHost)
+		req.Header.Add(storage.BUCKET, "")
 		return shardsRing.DoRequest(req)
 	}
+	reqHost, bucketName := removeBucketNameFromHost(reqHost)
+	if reqHost != "" && bucketName != "" {
+		shardsRing, ok = rg.multiCluters[reqHost]
+		if ok {
+			req.Header.Add(storage.HOST, reqHost)
+			req.Header.Add(storage.BUCKET, bucketName)
+			return shardsRing.DoRequest(req)
+		}
+	}
 	if rg.defaultRing != nil {
 		return rg.defaultRing.DoRequest(req)
 	}
 	return rg.getNoSuchDomainResponse(req), nil
 }
 
+
+func removeBucketNameFromHost(host string) (string, string){
+	splitted := strings.SplitN(host, ".", 2)
```

Comment:
This trick does not comply with bucket naming specs:
"Bucket names must be a series of one or more labels. Adjacent labels are separated by a single period (.)."
https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html

Reply:
Thanks, I didn't know that there can me multiple dots.

Reply:
Done